### DISCUSSION TITLE
Changing the JDK used by the maven (mvn) formula

### DISCUSSION THREAD

**Discussion by F0rce** [Homebrew/discussions#3865] [Everyday usage] [answered]
Hello everyone,

I installed maven via `brew install maven` quite a while ago. But just recently I saw that my build processes were different on Windows in Mac using maven. After further investigation I found out, that maven on my MacBook is using Java 17.

I tried two things, which didn't seem to work (`sudo nano /usr/local/Cellar/maven/3.8.6/bin/mvn`):
1. replaced JAVA_HOME with `JAVA_HOME="${JAVA_HOME:-$(/usr/libexec/java_home -v 11)}"` --> no change still using Java 17
2. replaced JAVA_HOME with `JAVA_HOME="${JAVA_HOME:-$(/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home)}"` --> no change still using Java 17

I am really out of ideas and would really appreciate some kind of help.

Mac OS version: 13.0 (22A380)
Brew: 3.6.8

Thanks in advance,
David

**Comment by SMillerDev** [comment]
What does `echo $JAVA_HOME` say?

**Reply by Bo98** [reply]
To add: the path in `/usr/local/Cellar/maven/3.8.6/bin/mvn` is a fallback, so you never need to edit it and it will just get overwritten on upgrades. It will always prioritise `$JAVA_HOME`. If you are getting Java 17 then that means you have set JAVA_HOME in your environment to that.

**Reply by F0rce** [reply]
Thanks for the replies so far :)

But unfortunately my $JAVA_HOME is unset and my `.zshrc` only sets PATH. But Java on the command line is working perfectly fine. Thats why I am here, because I am out of solutions.

<img width="363" alt="Bildschirm­foto 2022-11-04 um 08 10 36" src="https://user-images.githubusercontent.com/60381251/199914277-8fc1b272-2ee9-4567-9e56-a30d5e14bfaa.png">

**Reply by SMillerDev** [reply]
JAVA_HOME isn't set by default. You'll have to set it yourself in zshrc, pointing to the version you want to use.

**Reply by F0rce** [reply]
Thanks a lot didn't knew that. 

Just out of curiousity. How does my macbook know if I use Java, where it is located, if JAVA_HOME is not set ?

**Reply by SMillerDev** [reply]
Your MacBook doesn't, but individual programs might ship their own Java or a Java might be installed in a system default location.

**Comment by carlocab** [comment, marked as answer]
Try doing
```sh
export JAVA_HOME=/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home
```
or
```sh
export JAVA_HOME="$(/usr/libexec/java_home -v 11)"
```
before running `mvn`.

**Reply by F0rce** [reply]
Your solution worked by the way... I dont know how I didn't thought of setting the JAVA_HOME env...

But do I have to do this for every shell session ? Or should I then also place it in my zshrc ?
